examples/annotation.py

```diff
@@ -86,7 +88,7 @@ def multinomial(annotations):
     """
     This model corresponds to the plate diagram in Figure 1 of reference [1].
     """
-    num_classes = int(jnp.max(annotations)) + 1
+    num_classes = int(np.max(annotations)) + 1
```

Comment:
This is the main error: we can't cast `int` to a DeviceArray under jit anymore. Something likes those will cause errors now:
```
def f(x):
    return np.ones(10)[x]

jax.jit(f)(jnp.array([1, 2]))
```
or
```
def f(x):
    a = int(jnp.ones(()))
    return x + a

jax.jit(f)(jnp.array([1, 2]))
```

Reply:
So under jit, function arguments cannot be later used as concrete arrays?

Reply:
I think they are still concrete arrays. The new change is we can't apply numpy/python operators on DeviceArray under jit any more (previously, it is doable as long as that DeviceArray is constant - i.e. not derived from traced inputs). In addition, under jit, we can't use DeviceArray indexing for non-DeviceArray array. We have to do something like
```
def f(x):
    return jnp.ones(10)[x]
```
or
```
def f(x):
    return device_put(np.ones(10))[x]
```